### PR TITLE
Forward all props passed to downshift to the getStateAndHelpers function

### PR DESCRIPTION
**Pull request by Dajust**
This is the implementation of the idea detailed in this issue #225

### CODE REVIEW

**Comment by pbomb** [comment]
I'd prefer that we not make this change given that the solution to #225, like @kentcdodds mentioned, is (in most cases?) simple to handle in the consumer with an arrow function to re-pass the desired props to the event handlers. Additionally, -- and this is not a great argument, but -- spreading all the props adds a little extra overhead for all consumers who don't need this functionality. Handling this in a reusable component consumer could look something like this:

```js
class Autocomplete extends React.Component {
  render() {
    return (
      <Downshift name={this.props.name} onChange={ (value) => this.props.onChange(name, value) }>{ ... }</Downshift>
    );
  }
}
```

If we do decide to allow this change, then this Pull Request will also need to add tests and TypeScript type definitions. The type definition changes are more difficult than usual due to the dynamic nature of forwarding on all props.

**Comment by kentcdodds** [comment]
Well, we could avoid passing all the props if we pass the `id` in addition to the `itemToString`. Then @Dajust's desired outcome will be satisfied.

This would require a change to the section on props that I'll not inline.

File: README.md [hunk 1]
```diff
@@ -518,6 +517,10 @@ These are values that represent the current state of the downshift component.
 | `isOpen`           | `boolean`         | the menu open state                            |
 | `selectedItem`     | `any`             | the currently selected item input              |
 
+### props
+
+As a convenience, all props you pass to <Downshift /> are forwarded along to you. You can access all props via the `getStateAndHelpers` function.
```
Comment (kentcdodds) [inline README.md]: The `getStateAndHelpers` function is not exposed, so we shouldn't mention that. Just remove this sentence:

> You can access all props via the `getStateAndHelpers` function.

You could change this section to say:

```
### props

As a convenience, the `id` and `itemToString` props which you pass to `<Downshift />` are available here as well.
```

File: src/downshift.js [hunk 2]
```diff
@@ -332,7 +332,6 @@ class Downshift extends Component {
 
   getStateAndHelpers() {
     const {highlightedIndex, inputValue, selectedItem, isOpen} = this.getState()
-    const {itemToString} = this.props
```
Comment (kentcdodds) [inline src/downshift.js]: Let's restore this line and destructure out the `id` from here too

File: src/downshift.js [hunk 3]
```diff
@@ -352,6 +351,7 @@ class Downshift extends Component {
     } = this
     return {
       // prop getters
+      ...this.props,
```
Comment (kentcdodds) [inline src/downshift.js]: Let's remove this.

File: src/downshift.js [hunk 4]
```diff
@@ -369,7 +369,6 @@ class Downshift extends Component {
       setHighlightedIndex,
       clearSelection,
       clearItems,
-      itemToString,
```
Comment (kentcdodds) [inline src/downshift.js]: Let's restore this and add the `id` below it.

Then above this line let's leave a blank line and then a comment that says: `// props` like the other sections.

**Pull request merged**